imap_data_access/io.py

```diff
@@ -28,7 +28,7 @@ def download(file_path: str) -> Path:
         Path to the downloaded file
     """
     destination = imap_data_access.config["DATA_DIR"]
-    if "/" not in file_path:
+    if isinstance(file_path, str) and "/" not in file_path:
```

Comment:
will this only check if path is string or does it convert Path object to str for the check?

Reply:
It see if it is a string, then if it is a string checks if "/" is in that string, and if it isn't then it goes down this if-block. (we have only been given a filename, not the full path).

We could tighten this up if we wanted and remove this altogether and require the full path even for the download routines?

Reply:
I feel like it should something like:
```
if file_path is Path object, then convert to str. Then check for '/' in file_path.
else if file_path is str, then '/' in file_path
```
The issue was that download would fail in input is path object because we don't convert that to str and as a result `if "/" not in file_path:` check fails.

Reply:
I see. I just saw your reply. I think that makes sense.